File: TypeScriptTest/public_html/word.js
//////////////////////////////////////////////////////////////////////

var Word = (function () {
    "use strict";

    return chs.Class({
        
        static$: {
            horizontal: 0,
            vertical: 1
        },

        $: function (str, x, y, orientation, score) {
            this.str = str;
            this.x = x;
            this.y = y;
            this.orientation = orientation;
            this.score = score;
            this.index = 0;
            this.listNode = chs.List.Node(this);
        },

        toString: function () {
            return this.str + "(" + this.score.toString() + ")";
        }
    });

}());

//////////////////////////////////////////////////////////////////////

File: TypeScriptTest/web_service/word.py
from letters import Letters

class Word:

    horizontal = 0
    vertical = 1

    def __init__(word, x, y, direction, score):
        self.word = word
        self.x = x
        self.y = y
        self.direction = direction
        self.score = score

    @staticmethod
    def compare(x, y):
        if x.score < y.score:
            return -1
        if x.score > y.score:
            return 1
        if len(x.word) < len(y.word):
            return -1
        if len(x.word) > len(y.word):
            return 1
        if x.word < y.word:
            return -1
        if x.word > y.word:
            return 1
        return 0
